scapy/packet.py

```diff
@@ -1012,10 +1012,12 @@ def haslayer(self, cls):
                         return ret
         return self.payload.haslayer(cls)
 
-    def getlayer(self, cls, nb=1, _track=None, _subclass=False, **flt):
+    def getlayer(self, cls, nb=1, _track=None, _subclass=None, **flt):
         """Return the nb^th layer that is an instance of cls, matching flt
 values.
         """
+        if _subclass is None:
+            _subclass = isinstance(self, PacketMatchSubclass) or None
```

Comment:
I don't think we need the `or None` here. After this, `_subclass` should be either `True` or `False`. This won't change how things work, but is clearer IMO.
```Suggestion
            _subclass = isinstance(self, PacketMatchSubclass)
```

Reply:
Without the `or None`, `_subclass` is set to False and won’t be executed for the payload